noir-projects/noir-contracts/contracts/key_registry_contract/src/main.nr

```diff
@@ -51,19 +51,29 @@ contract KeyRegistry {
         let npk_m_y_registry = storage.npk_m_y_registry.at(address);
         let ivpk_m_x_registry = storage.ivpk_m_x_registry.at(address);
         let ivpk_m_y_registry = storage.ivpk_m_y_registry.at(address);
-        // let ovpk_m_x_registry = storage.ovpk_m_x_registry.at(address);
-        // let ovpk_m_y_registry = storage.ovpk_m_y_registry.at(address);
-        // let tpk_m_x_registry = storage.tpk_m_x_registry.at(address);
-        // let tpk_m_y_registry = storage.tpk_m_y_registry.at(address);
 
         npk_m_x_registry.schedule_value_change(keys.npk_m.x);
         npk_m_y_registry.schedule_value_change(keys.npk_m.y);
         ivpk_m_x_registry.schedule_value_change(keys.ivpk_m.x);
         ivpk_m_y_registry.schedule_value_change(keys.ivpk_m.y);
-        // Commented out as we hit the max enqueued public calls limit when not done so
-        // ovpk_m_x_registry.schedule_value_change(ovpk_m.x);
-        // ovpk_m_y_registry.schedule_value_change(ovpk_m.y);
-        // tpk_m_x_registry.schedule_value_change(tpk_m.x);
-        // tpk_m_y_registry.schedule_value_change(tpk_m.y);
+
+        // Note: This is a hack to get around MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_CALL limit
+        KeyRegistry::at(context.this_address())._register_remaining(address, keys.ovpk_m, keys.tpk_m).call(&mut context);
+    }
+
+    #[aztec(public)]
+    #[aztec(internal)]
+    fn _register_remaining(address: AztecAddress, ovpk_m: GrumpkinPoint, tpk_m: GrumpkinPoint) {
+        // We don't need to do any checks here because they were already performed in register(...) and this function is
+        // internal
+        let ovpk_m_x_registry = storage.ovpk_m_x_registry.at(address);
+        let ovpk_m_y_registry = storage.ovpk_m_y_registry.at(address);
+        let tpk_m_x_registry = storage.tpk_m_x_registry.at(address);
+        let tpk_m_y_registry = storage.tpk_m_y_registry.at(address);
+
+        ovpk_m_x_registry.schedule_value_change(ovpk_m.x);
+        ovpk_m_y_registry.schedule_value_change(ovpk_m.y);
+        tpk_m_x_registry.schedule_value_change(tpk_m.x);
+        tpk_m_y_registry.schedule_value_change(tpk_m.y);
```

Comment:
With this added down here, we could also expose the function at the getter higher up? The reason was that it would never be set but it can be now 🤔

Reply:
Good idea, will do in a follow up cleanup PR.